ca/server.go

```diff
@@ -569,9 +590,15 @@ func (s *Server) UpdateRootCA(ctx context.Context, cluster *api.Cluster) {
 	// ExternalCA interface that has different implementations for
 	// different CA types. At the moment, only CFSSL is supported.
 	var cfsslURLs []string
-	for _, ca := range cluster.Spec.CAConfig.ExternalCAs {
-		if ca.Protocol == api.ExternalCA_CAProtocolCFSSL {
-			cfsslURLs = append(cfsslURLs, ca.URL)
+	for _, extCA := range cluster.Spec.CAConfig.ExternalCAs {
+		// We want to support old external CA specifications which did not have a CA cert.  If there is no cert specified,
+		// we assume it's the old cert
+		certForExtCA := extCA.CACert
+		if certForExtCA == nil {
+			certForExtCA = rCA.CACert
+		}
+		if extCA.Protocol == api.ExternalCA_CAProtocolCFSSL && bytes.Equal(certForExtCA, wantedExternalCACert) {
+			cfsslURLs = append(cfsslURLs, extCA.URL)
```

Comment:
It may be noisy, just because we update every time the cluster object is changed, as opposed to specifically when the `RootCA` object or external CAs are changed.  I wonder if we should copy the `RootCA` object and `externalCA` objects to the server so that we can skip doing any changes if the objects themselves haven't changed...